game.cpp

```diff
@@ -36,4 +35,62 @@ Game::Game()
     playerA = 5.25f;
 }
 
-Game::~Game() {}
+bool Game::IsWall(uint8_t tileX, uint8_t tileY)
+{
+    if (tileX > MAP_X - 1 || tileY > MAP_Y - 1) {
+        return true;
+    }
+    return LOOKUP8(g_map, (tileX >> 3) + (tileY << (MAP_XS - 3))) &
+           (1 << (8 - (tileX & 0x7)));
+}
+
+void Game::CollideWithMap()
+{
+    float offsetX, offsetY, mapX, mapY;
+    bool collideX, collideY;
+    offsetX = modff(playerX, &mapX);
+    offsetY = modff(playerY, &mapY);
+
+    // collide with four neighboring wall planes
+    // assuming PLAYER_RADIUS < 0.5.
```

Comment:
You can use `static_assert` for the valid range, performing compile-time assertion checking.